contracts/Controller.sol

```diff
@@ -37,4 +36,42 @@ contract Controller is Pausable, IController {
     function updateController(bytes32 _id, address _controller) external onlyOwner returns (bool) {
         return IManager(registry[_id]).setController(_controller);
     }
+
+    /*
+     * @dev Add a function permission for a contract
+     * @param _src Source address (caller)
+     * @param _target Target address (contract called)
+     * @param _sig Function signature at target address
+     */
+    function addPermission(address _src, address _target, bytes4 _sig) external onlyOwner {
```

Comment:
if we update a contract, and the signature changes, then we have to be sure to update permissions as well to use the new signature, right? Can we do this atomically? Or perhaps we can add the permission first before deploying the update?